tools/nr_from_src.py

```diff
@@ -22,32 +24,39 @@ def load_table(path: str, arches: Set[str]):
                 yield (name, int(nr))
 
 def eval_expr(expr: str) -> int:
-    ma = SIMPLE_MATH.match(expr)
-    if ma:
-        return int(ma.group(1)) + int(ma.group(2))
-    raise Exception('"{}" looks like an expression, but not a supported one'.format(expr))
+    if not SIMPLE_MATH.match(expr):
+        raise Exception('"{}" looks like an expression, but not a supported one'.format(expr))
+    return sum(int(x.group(0), 0) for x in NUMBER.finditer(expr))
 
 
-def load_headers(names: Iterable[str], arch: str, extra: str = ''):
+def load_headers(names: Iterable[Tuple[str, str]], arch: str, extra: str = ''):
     with tempfile.NamedTemporaryFile(mode='w+', suffix='.h') as f:
-        f.write(extra)
-        f.write('\n')
-        f.write('#include <asm/unistd.h>\n')
-        for name in names:
-            f.write('gen_nr {0} __NR_{0}\n'.format(name))
-        f.flush()
-        lines = subprocess.check_output(['gcc', '-nostdinc',
-            '-I', '{}/arch/{}/include/uapi'.format(linux_path, arch),
-            '-I', '{}/include'.format(linux_path),
-            '-P', # don't include line number markers, which make the output annoying to parse
-            '-E', # only preprocess, don't compile
-            f.name]).decode('utf-8').split('\n')
+        with tempfile.TemporaryDirectory() as temp_include_dir:
+            os.mkdir('{}/asm'.format(temp_include_dir))
+            with open('{}/asm/unistd-eabi.h'.format(temp_include_dir), 'w'):
```

Comment:
is this like `touch unistd-eabi.h`? Could you add a comment about why this is needed?